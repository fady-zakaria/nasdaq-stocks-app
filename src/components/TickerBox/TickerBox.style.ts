import { Dimensions, StyleSheet } from "react-native";
import { Colors } from "../../utils/Colors";

const { width } = Dimensions.get('window');

export const styles = StyleSheet.create({
    container:{
        display: 'flex',
        flexDirection: 'column',
        justifyContent: 'flex-start',
        alignItems: 'center',
        backgroundColor: Colors.primaryColor,
        borderRadius: 15,
        paddingHorizontal: 10,
        paddingVertical: 20,
        marginTop: 10,
        overflow: 'hidden',
        borderWidth: 0.5,
        borderColor: Colors.borderColor,
        width: (width - 60) / 2,
    },
    tickerText: {
        color: Colors.primaryText,
        marginVertical:10,
        fontWeight:'700',
        textAlign: "center",
    },
    stockName: {
        color: Colors.secondaryText,
        fontSize: 15,
        textAlign: "center",
        width: 100,
    },
    img: {
        width: 50,
        height: 50,
        borderRadius: 50,
    },
})