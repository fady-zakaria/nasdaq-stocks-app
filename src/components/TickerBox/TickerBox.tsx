import {View, Text, Image, ImageSourcePropType} from 'react-native';
import React, {FC} from 'react';
import { styles } from './TickerBox.style';
import { TouchableOpacity } from 'react-native-gesture-handler';

interface Iprops {
  src?: ImageSourcePropType;
  name?: string;
  ticker?: string;
}

const TickerBox: FC<Iprops> = ({src, name, ticker}) => {
  return (
    <TouchableOpacity style={styles.container}>
       <Image
          style={styles.img}
          source={{uri: 'https://picsum.photos/200'}}
        />
          <Text style={styles.tickerText} numberOfLines={2}>
          {ticker}
          </Text>
            <Text style={styles.stockName} numberOfLines={2}>
            {name}
          </Text>
    </TouchableOpacity>
  );
};

export default TickerBox;
