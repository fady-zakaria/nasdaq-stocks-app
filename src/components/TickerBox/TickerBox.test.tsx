import React from 'react';
import renderer from 'react-test-renderer';
import TickerBox from './TickerBox';

describe('TickerBox', () => {
  it('renders correctly with given props', () => {
    const tree = renderer.create(
      <TickerBox
        src={{ uri: 'https://picsum.photos/200' }}
        name="Stock Name"
        ticker="SN"
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with default image source', () => {
    const tree = renderer.create(
      <TickerBox
        name="Stock Name"
        ticker="SN"
      />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

});