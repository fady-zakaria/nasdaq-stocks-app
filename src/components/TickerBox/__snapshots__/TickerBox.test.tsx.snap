// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TickerBox renders correctly with default image source 1`] = `
<RNGestureHandlerButton
  collapsable={false}
  delayLongPress={600}
  enabled={true}
  exclusive={true}
  handlerTag={2}
  handlerType="NativeViewGestureHandler"
  onGestureEvent={[Function]}
  onGestureHandlerEvent={[Function]}
  onGestureHandlerStateChange={[Function]}
  onHandlerStateChange={[Function]}
  rippleColor={0}
  touchSoundDisabled={false}
>
  <View
    accessible={true}
    collapsable={false}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#242639",
        "borderColor": "#323443",
        "borderRadius": 15,
        "borderWidth": 0.5,
        "display": "flex",
        "flexDirection": "column",
        "justifyContent": "flex-start",
        "marginTop": 10,
        "opacity": 1,
        "overflow": "hidden",
        "paddingHorizontal": 10,
        "paddingVertical": 20,
        "width": 345,
      }
    }
  >
    <Image
      source={
        {
          "uri": "https://picsum.photos/200",
        }
      }
      style={
        {
          "borderRadius": 50,
          "height": 50,
          "width": 50,
        }
      }
    />
    <Text
      numberOfLines={2}
      style={
        {
          "color": "#fcfcfc",
          "fontWeight": "700",
          "marginVertical": 10,
          "textAlign": "center",
        }
      }
    >
      SN
    </Text>
    <Text
      numberOfLines={2}
      style={
        {
          "color": "#abacb3",
          "fontSize": 15,
          "textAlign": "center",
          "width": 100,
        }
      }
    >
      Stock Name
    </Text>
  </View>
</RNGestureHandlerButton>
`;

exports[`TickerBox renders correctly with given props 1`] = `
<RNGestureHandlerButton
  collapsable={false}
  delayLongPress={600}
  enabled={true}
  exclusive={true}
  handlerTag={1}
  handlerType="NativeViewGestureHandler"
  onGestureEvent={[Function]}
  onGestureHandlerEvent={[Function]}
  onGestureHandlerStateChange={[Function]}
  onHandlerStateChange={[Function]}
  rippleColor={0}
  touchSoundDisabled={false}
>
  <View
    accessible={true}
    collapsable={false}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#242639",
        "borderColor": "#323443",
        "borderRadius": 15,
        "borderWidth": 0.5,
        "display": "flex",
        "flexDirection": "column",
        "justifyContent": "flex-start",
        "marginTop": 10,
        "opacity": 1,
        "overflow": "hidden",
        "paddingHorizontal": 10,
        "paddingVertical": 20,
        "width": 345,
      }
    }
  >
    <Image
      source={
        {
          "uri": "https://picsum.photos/200",
        }
      }
      style={
        {
          "borderRadius": 50,
          "height": 50,
          "width": 50,
        }
      }
    />
    <Text
      numberOfLines={2}
      style={
        {
          "color": "#fcfcfc",
          "fontWeight": "700",
          "marginVertical": 10,
          "textAlign": "center",
        }
      }
    >
      SN
    </Text>
    <Text
      numberOfLines={2}
      style={
        {
          "color": "#abacb3",
          "fontSize": 15,
          "textAlign": "center",
          "width": 100,
        }
      }
    >
      Stock Name
    </Text>
  </View>
</RNGestureHandlerButton>
`;
